### PR TITLE
Referral Form Functional Stored Procedures

### LINKED ISSUE #84: Referral Stored Procedures
We are working on getting the stored procedures to properly insert into the database. Currently working through a number of bugs.

### PR DESCRIPTION
**Pull request by christianmenk**
As mentioned in the title, all of the procedures are working and have been tested in the database.

PeopleInsert, addAgencyReferral, insert statements for phone numbers,
agencyMemberInsert, and createFamilyMember are all called. Methods for
executing these stored procedures should be documented within the code.

Fixes #84  

Thanks guys!

### CODE REVIEW

File: views/forms/referral_form.php [hunk 1]
```diff
@@ -223,21 +224,164 @@
...
             $office_contact_date,
-            '$office_means',
+            $office_means,
             $office_initial_date,
-            '$office_location',
-            '$comments',
-            1)); */
+            $office_location,
+            $comments,
+            $employeeID));
```
Comment (blackjack26) [inline views/forms/referral_form.php:231]: You do not actually check to see if this stored procedure actually executed properly. I didn't check for this on the self-referral but, if this query fails, it will say there is a form success when there actually isn't.

You should check the result like the following:
```php
if ($result) {
    $state = pg_result_error_field($result, PGSQL_DIAG_SQLSTATE);
    if ($state != 0) {
        // TODO: actually handle error don't kill the program and display the error (just an example)
        die(pg_result_error($result));
    }
}
```

The only reason I mention this is because I get an error when I have nothing populated in my `ZipCodes` table.
> ERROR: null value in column "zipcode" violates not-null constraint DETAIL: Failing row contains (null, null, null). CONTEXT: SQL statement "INSERT INTO ZipCodes VALUES (zip, city, CAST(state AS STATES))" PL/pgSQL function zipcodesafeinsert(integer,text,states) line 8 at SQL statement SQL statement "SELECT zipCodeSafeInsert(addAgencyReferral.zipCode, city, state)" PL/pgSQL function addagencyreferral(integer,date,integer,text,text,integer,text,states,text,boolean,text,date,text,boolean,boolean,boolean,boolean,boolean,boolean,boolean,boolean,date,text,timestamp without time zone,text,text,integer) line 51 at PERFORM

If you are going to default zip to 12601, you should also default the city and state to Poughkeepsie and New York (assuming the zip is not set to something else). This will at least not throw the error above. But if I just type in a zip code and no state or city and that zip code is not in the database, this error will occur.

---

**TLDR**: If the zip code is set by the user, you *need* to require the state and city. If the zip code is not set by the user, you should default the city and state to Poughkeepsie and NY. The same goes for any combination of zip, city, and state. If any one of those is set, the other 2 need to be required. Otherwise, the `zipCodeSafeInsert()` function will fail.

Comment (blackjack26) [reply views/forms/referral_form.php:231]: I'm pretty sure this is the same case for the self-referral but I missed it, if you can change it there too that'd be ideal. You need to validate that the query was successful before moving on. It won't automatically throw an error for you.

@sbohner14335 
For self-referral I get the error:
> ERROR: invalid input syntax for type date: ""

When submitting the form with just a first and last name.

**Review by blackjack26 — CHANGES_REQUESTED** [review]

File: views/forms/referral_form.php (hunk 1 above)
Comment (sbohner14335) [reply views/forms/referral_form.php:231]: So you're right about the error handling as well as the zipCodeSafeInsert() we will definitely do that. 

As for the self-referral form, the db team @jrands knows about the DOB error and I'm pretty sure that they are going to fix that on their end, since it shouldn't be a required field in the form itself.

File: views/forms/referral_form.php [hunk 2]
```diff
@@ -379,9 +389,9 @@
 
 
 
-    $_SESSION['form-type'] = $form_type;
-    header("Location: /form-success");
-    die();
+//    $_SESSION['form-type'] = $form_type;
+//    header("Location: /form-success");
+//    die();
```
Comment (blackjack26) [inline views/forms/referral_form.php]: Uncomment this

**Review by blackjack26 — APPROVED** [review]
